packages/gestalt-datepicker/src/DatePicker/InternalDatePicker.js

```diff
@@ -38,26 +38,21 @@ const InternalDatePickerWithForwardRef: AbstractComponent<Props, HTMLInputElemen
     rangeSelector,
     rangeStartDate,
     selectLists,
-    value: dateValue,
+    value: controlledValue,
   }: Props,
   ref,
 ): Element<'div'> {
   const innerInputRef = useRef<null | HTMLInputElement>(null);
   useImperativeHandle(ref, () => innerInputRef.current);
 
-  const [selected, setSelected] = useState<?Date>(dateValue);
+  const [uncontrolledValue, setUncontrolledValue] = useState<?Date>(null);
```

Comment:
nit: I think we should use the original variable naming, since this feels very generic and value seems like it can mean anything

Reply:
Disagree. 

controlled and uncontrolled components have `selected` values or dates. 

Here i'm trying to differentiate the values used when the component works as controlled  and uncontrolled

Reply:
and it's naming. I could go _uncontrolledSelectedValue_ and _controlledSelectedValue_.

value is the standard for the value we pass to fields